ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -378,6 +378,44 @@ protected void updateResolvedContainers(Account account, Collection<Container> r
     }
   }
 
+  @Override
+  public Container getContainer(String accountName, String containerName) throws AccountServiceException {
+    Account account = getAccountByName(accountName);
+    if (account == null) {
+      return null;
+    }
+    Container container = account.getContainerByName(containerName);
+    if (container == null) {
+      // If container is not present in the account, refresh cache and retry
+      try {
+        fetchAndUpdateCache();
+        container = getAccountByName(accountName).getContainerByName(containerName);
```

Comment:
If container is still null here, I would cache the account-container combo in a LRU cache, and check membership in that cache before querying the DB.